pkg/function/delete_csi_snapshot_test.go

```diff
@@ -0,0 +1,146 @@
+
+func (testSuite *DeleteCSISnapshotTestSuite) SetUpSuite(c *C) {
+	testSuite.volumeSnapshotClass = DeleteCSISnapshotSnapshotClass
+	testSuite.storageClass = DeleteCSISnapshotStorageClass
+	testSuite.pvcName = DeleteCSISnapshotOriginalPVCName
+	testSuite.newPVCName = DeleteCSISnapshotNewPVCName
+	testSuite.snapName = DeleteCSISnapshotSnapshotName
+	testSuite.namespace = DeleteCSISnapshotTestNamespace
+}
+
+func (testSuite *DeleteCSISnapshotTestSuite) TestDeleteCSISnapshot(c *C) {
+	for _, apiResourceList := range []*metav1.APIResourceList{
+		{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "VolumeSnapshot",
+				APIVersion: "v1alpha1",
+			},
+			GroupVersion: "snapshot.storage.k8s.io/v1alpha1",
+		},
+		{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "VolumeSnapshot",
+				APIVersion: "v1beta1",
+			},
+			GroupVersion: "snapshot.storage.k8s.io/v1beta1",
+		},
+		{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "VolumeSnapshot",
+				APIVersion: "v1",
+			},
+			GroupVersion: "snapshot.storage.k8s.io/v1",
+		},
+	} {
+		fakeCli := fake.NewSimpleClientset()
+		fakeCli.Resources = []*metav1.APIResourceList{apiResourceList}
+
+		_, err := fakeCli.CoreV1().Namespaces().Create(context.TODO(), &v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: testSuite.namespace}}, metav1.CreateOptions{})
+		c.Assert(err, IsNil)
+
+		scheme := runtime.NewScheme()
+		fakeSnapshotter, err := snapshot.NewSnapshotter(fakeCli, dynfake.NewSimpleDynamicClient(scheme))
+		c.Assert(err, IsNil)
+
+		originalPVC := &v1.PersistentVolumeClaim{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: testSuite.pvcName,
+			},
+			Spec: v1.PersistentVolumeClaimSpec{
+				StorageClassName: &testSuite.storageClass,
+				AccessModes:      []v1.PersistentVolumeAccessMode{v1.ReadWriteOnce},
+				Resources: v1.ResourceRequirements{
+					Requests: v1.ResourceList{
+						v1.ResourceStorage: resource.MustParse("1Gi"),
+					},
+				},
+			},
+		}
+		_, err = fakeCli.CoreV1().PersistentVolumeClaims(testSuite.namespace).Create(context.TODO(), originalPVC, metav1.CreateOptions{})
+		c.Assert(err, IsNil)
+
+		err = fakeSnapshotter.Create(context.Background(), testSuite.snapName, testSuite.namespace, testSuite.pvcName, &testSuite.volumeSnapshotClass, false, nil)
+		c.Assert(err, IsNil)
+
+		vs, err := fakeSnapshotter.Get(context.Background(), testSuite.snapName, testSuite.namespace)
+		c.Assert(err, IsNil)
+		c.Assert(vs.Name, Equals, testSuite.snapName)
+
+		restoreArgs := restoreCSISnapshotArgs{
+			Name:         testSuite.snapName,
+			PVC:          testSuite.newPVCName,
+			Namespace:    testSuite.namespace,
+			StorageClass: testSuite.storageClass,
+			RestoreSize:  originalPVC.Spec.Resources.Requests.Storage(),
+			AccessModes:  originalPVC.Spec.AccessModes,
+			Labels:       nil,
+		}
+		newPVC := newPVCManifest(restoreArgs)
+		_, err = fakeCli.CoreV1().PersistentVolumeClaims(restoreArgs.Namespace).Create(context.TODO(), newPVC, metav1.CreateOptions{})
```

Comment:
instead of creating the context at places, let's create on and use that at other places.

Reply:
Done